src/v/storage/batch_cache.cc

```diff
@@ -236,12 +299,12 @@ void batch_cache_index::truncate(model::offset offset) {
     if (auto it = find_first(offset); it != _index.end()) {
         // rule out if possible, otherwise always be pessimistic
         if (
-          it->second && it->second->valid()
-          && !it->second->batch().contains(offset)) {
+          it->second.range() && it->second.valid()
+          && !it->second.batch().contains(offset)) {
```

Comment:
does this `it->second.batch()` materialize the batch to check `contains`? should we add an interface to grab only the header?

Reply:
great idea